galsim/config/stamp_ring.py

```diff
@@ -67,7 +67,7 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
 
         # Now go on and do the base class setup.
         ignore = ignore + list(req) + list(opt)
-        return super(self.__class__, self).setup(config, base, xsize, ysize, ignore, logger)
+        return super(RingBuilder, self).setup(config, base, xsize, ysize, ignore, logger)
```

Comment:
Yes.  If you subclass from RingBuilder (which I did), then `self.__class__` is the subtype, not RingBuidler, and `super(self.__class__, self)` is `RingBuilder`, not `StampBuilder`, which leads to an infinite loop.

I had seen the trick about writing `super(self.__class__, self)` on StackOverflow somewhere, and I thought it was a clever way to make sure you do the right thing.  But it turns out it breaks in this kind of use case, so I think writing out the actual class name is better practice in general.